Tests/Integration/JoinSlowTwinUpdateTests.cs

```diff
@@ -41,30 +41,29 @@ public async Task When_First_Join_Fails_Due_To_Slow_Twin_Update_Retry_Second_Att
                 .ReturnsAsync(twin);
 
             // Device twin will be updated
-            AppSessionKey? afterJoin1AppSKey = null;
-            NetworkSessionKey? afterJoin1NwkSKey = null;
-            string afterJoin1DevAddr = null;
             AppSessionKey? afterJoin2AppSKey = null;
             NetworkSessionKey? afterJoin2NwkSKey = null;
             string afterJoin2DevAddr = null;
 
-            LoRaDeviceClient.Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>()))
-                .ReturnsAsync(true, TimeSpan.FromSeconds(10))
-                .Callback<TwinCollection>((updatedTwin) =>
+            var invocations = 0;
+            LoRaDeviceClient.Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>(), It.IsAny<CancellationToken>()))
+                .Returns<TwinCollection, CancellationToken>(async (updatedTwin, token) =>
                 {
-                    afterJoin1AppSKey = AppSessionKey.Parse(updatedTwin[TwinProperty.AppSKey].Value);
-                    afterJoin1NwkSKey = NetworkSessionKey.Parse(updatedTwin[TwinProperty.NwkSKey].Value);
-                    afterJoin1DevAddr = updatedTwin[TwinProperty.DevAddr];
-
-                    // update setup for no delay
-                    LoRaDeviceClient.Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>()))
-                        .ReturnsAsync(true)
-                        .Callback<TwinCollection>((updatedTwin2) =>
-                        {
-                            afterJoin2AppSKey = AppSessionKey.Parse(updatedTwin2[TwinProperty.AppSKey].Value);
-                            afterJoin2NwkSKey = NetworkSessionKey.Parse(updatedTwin2[TwinProperty.NwkSKey].Value);
-                            afterJoin2DevAddr = updatedTwin2[TwinProperty.DevAddr];
-                        });
+                    if (invocations == 0)
+                    {
+                        invocations++;
+                        await Task.Delay(7_000, CancellationToken.None);
```

Comment:
shouldn't we pass the token here?

Reply:
then you don't need the code below.

Reply:
In the test I want to actually assert that the first request is cancelled after 7 seconds while the second join request is just going through properly. Added a comment and the Sequence setup in 9189435